pom.xml

```diff
@@ -123,11 +123,45 @@
           <artifactId>maven-bundle-plugin</artifactId>
           <version>3.3.0</version>
           <extensions>true</extensions>
+          <dependencies>
+            <dependency>
+              <!-- Required to make the plugin work on JDK9. See https://issues.apache.org/jira/browse/FELIX-5698. -->
+              <groupId>biz.aQute.bnd</groupId>
+              <artifactId>biz.aQute.bndlib</artifactId>
+              <version>3.5.0</version>
+            </dependency>
+          </dependencies>
+
         </plugin>
       </plugins>
     </pluginManagement>
     <plugins>
       <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-compiler-plugin</artifactId>
+        <executions>
+          <execution>
+            <id>default-compile</id>
+            <configuration>
+              <!-- compile everything to ensure module-info contains right entries -->
+              <release>9</release>
```

Comment:
I was under the impression that if you compile with release=8 and there is a module-info.java, it actually compiles that too and puts it in the jar and JDK 8 ignores it (it was intentionally build that way) and JDK 9 picks it up as if the jar was build for release 9.
(This is what I understood from the fosdem talk of Mark Reinhold earlier this year.)

That being said, I suppose this info is useless as we're also targetting 6 or 7?

Reply:
The compilation needs to happen in two steps (yeah, that is not ideal, but it is the [way recommended by maven-compiler plugin](https://maven.apache.org/plugins/maven-compiler-plugin/examples/module-info.html). What basically happens is that in the first run compiles everything with -release 9, including the module-info.java. The second run then uses source=6 + target=6 (which should probably be replace by -release as well) to re-compile everything, but excluding the java-module.info.

It is not possible compile java-module.info with -release=8. However, in order to compile module-info.java the JDK9's javac also need to have info about all the sources files. 

This effectively makes the compilation twice as long. For a small project like this, it is fine, but for bigger projects this may actually be an minor issue.